src/ui/base-page-storyEdit.js

```diff
@@ -384,55 +452,72 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
     };
 
     /**
-     * Submits the editor form to the server
+     * Saves the story to the server
      *
-     * @param {jQuery} storyEditorForm - the Editor UI's HTML form element
+     * @param {String} storySaveUrl - the server URL at which to save a story
      * @param {Object} storyModel - the model of the story to save
-     * @param {Object} errorEvent - an event to fire on errors
-     * @param {String} storyAutosaveKey - the key at which story content is saved
+     *
+     * @return {jqXHR} - the jqXHR for the server request
      */
-    sjrk.storyTelling.base.page.storyEdit.submitStory = function (storyEditorForm, storyModel, errorEvent, storyAutosaveKey) {
-        storyEditorForm.attr({
-            action: "/stories/",
-            method: "post",
-            enctype: "multipart/form-data"
+    sjrk.storyTelling.base.page.storyEdit.saveStoryToServer = function (storySaveUrl, storyModel) {
+        return $.ajax({
+            url         : storySaveUrl,
+            data        : JSON.stringify(storyModel),
+            cache       : false,
+            contentType : "application/json",
+            processData : false,
+            type        : "POST"
         });
+    };
 
-        // This is the easiest way to be able to submit form
-        // content in the background via ajax
-        var formData = new FormData(storyEditorForm[0]);
+    /**
+     * @callback publishSuccessCallback - a callback to call after successful publishing
+     * @param {String} storyId - the ID of the story to which the user will be redirected
+     * @param {String} viewPageUrl - the URL for the story View page
+     */
 
-        // Stores the entire model as a JSON string in one
-        // field of the multipart form
-        var modelAsJSON = JSON.stringify(storyModel);
-        formData.append("model", modelAsJSON);
+    /**
+     * Sets a story to "published" and redirects the user to the new story page
+     *
+     * @param {Component} storyEditPage - an instance of `sjrk.storyTelling.base.page.storyEdit`
+     * @param {String} viewPageUrl - the URL for the story View page
+     * @param {Component} story - an instance of `sjrk.storyTelling.story`
+     * @param {publishSuccessCallback} successCallback - to call on successful publish
+     * @param {Object} errorEvent - the event to be fired in case of an error
+     */
+    sjrk.storyTelling.base.page.storyEdit.publishStory = function (storyEditPage, viewPageUrl, story, successCallback, errorEvent) {
+        // set the publish timestamp to now and the flag to "true"
+        story.applier.change("timestampPublished", new Date().toISOString());
+        story.applier.change("published", true);
```

Comment:
Consider to optimize the change request to:

```suggestion
story.applier.change("", {
    "timestampPublished": new Date().toISOString(),
    "published": true
});
```

The benefit is when there's a model listener listening to the root of the model, this listener won't be fired twice.